File: src/main/java/no/nav/syfo/kafka/producer/TredjepartsvarselProducer.java
package no.nav.syfo.kafka.producer;

import no.nav.syfo.kafka.producer.model.KTredjepartsvarsel;
import org.slf4j.Logger;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

import static java.util.UUID.randomUUID;
import static org.slf4j.LoggerFactory.getLogger;

@Component
public class TredjepartsvarselProducer {

    private static final Logger log = getLogger(TredjepartsvarselProducer.class);

    private KafkaTemplate<String, Object> kafkaTemplate;

    public final String TREDJEPARTSVARSEL_TOPIC = "aapen-syfo-tredjepartsvarsel-v1";

    public TredjepartsvarselProducer(
            KafkaTemplate<String, Object> kafkaTemplate
    ) {
        this.kafkaTemplate = kafkaTemplate;
    }

    public void sendTredjepartsvarselvarsel(KTredjepartsvarsel kTredjepartsvarsel) {
        try {
            kafkaTemplate.send(
                    TREDJEPARTSVARSEL_TOPIC,
                    randomUUID().toString(),
                    kTredjepartsvarsel).get();
            log.info("Legger tredjepartsvarsel med ressursID {} på kø for aktor {}", kTredjepartsvarsel.ressursId(), kTredjepartsvarsel.aktorId());
        } catch (Exception e) {
            log.error("Feil ved sending av oppgavevarsel", e);
            throw new RuntimeException("Feil ved sending av oppgavevarsel", e);
        }
    }
}

File: src/main/kotlin/no/nav/syfo/varsel/TredjepartsvarselProducer.kt
package no.nav.syfo.varsel

import no.nav.syfo.kafka.producer.model.KTredjepartsvarsel
import org.slf4j.LoggerFactory
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component
import java.util.*
import javax.inject.Inject

@Component
class TredjepartsvarselProducer @Inject constructor(
        private val kafkaTemplate: KafkaTemplate<String, Any>
) {
    fun sendTredjepartsvarselvarsel(kTredjepartsvarsel: KTredjepartsvarsel) {
        try {
            kafkaTemplate.send(
                    TREDJEPARTSVARSEL_TOPIC,
                    UUID.randomUUID().toString(),
                    kTredjepartsvarsel).get()
            log.info("Legger tredjepartsvarsel med ressursID {} på kø for aktor {}", kTredjepartsvarsel.ressursId, kTredjepartsvarsel.aktorId)
        } catch (e: Exception) {
            log.error("Feil ved sending av oppgavevarsel", e)
            throw RuntimeException("Feil ved sending av oppgavevarsel", e)
        }
    }

    companion object {
        private val log = LoggerFactory.getLogger(TredjepartsvarselProducer::class.java)
        const val TREDJEPARTSVARSEL_TOPIC = "aapen-syfo-tredjepartsvarsel-v1"
    }
}
